File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    rank=[]
    
    for i in range(0,len(score)) :
        if len(rank) < k :
            rank.append(score[i])
            rank.sort()
            
        else : #len(rank)==k :
            if rank[0] < score[i] :
                rank[0] = score[i]
                rank.sort()
            
        answer.append(rank[0])
    return answer

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.ArrayList;
import java.util.Collections;


class Solution {
    public int[] solution(int k, int[] score) {
        ArrayList<Integer> answer = new ArrayList<>();
        int[] arr= new int[score.length];
        
        for(int i =0; i<score.length; i++){
            answer.add(score[i]);
            Collections.sort(answer, Collections.reverseOrder());
            
            if(answer.size() < k){
                arr[i] = answer.get(i);   
            } else{
                arr[i] = answer.get(k-1);
            }
        }
        return arr;
    }
}